build/common.props

```diff
@@ -16,4 +16,12 @@
     <VersionSuffix Condition="'$(VersionSuffix)'!='' AND '$(BuildNumber)' != ''">$(VersionSuffix)-$(BuildNumber)</VersionSuffix>
   </PropertyGroup>
 
-</Project>
\ No newline at end of file
+  <!-- resolve using git.exe because the GetGitCommitInfoTask can't be loaded before it is compiled -->
+  <!-- TODO set this in repo.targets so it applies to the entire build -->
+  <Target Name="ResolveCommitHashWithExe" Condition="'$(CommitHash)'==''" BeforeTargets="PrepareForBuild">
+    <Exec Command="git rev-parse HEAD" ConsoleToMSBuild="true" EchoOff="true" Condition="'$(CommitHash)'==''">
```

Comment:
Yeah, I considered that too. But decided to keep using the task for the reason you mentioned.
Plus, we don't want to require the source code to be in git either. This implementation will warn but not fail if git is unavailable.